util/opentitan-pgm-fpga/opentitan-pgm-fpga

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""Program an FPGA with a bitstream
+"""
+import argparse
+import os
+import subprocess
+import sys
+
+
+def program_bitstream_with_vivado(part,
+                                  bitstream,
+                                  hw_target="",
+                                  jtag_frequency=""):
+    tcl_file = os.path.join(os.path.dirname(__file__), 'vivado_pgm.tcl')
+    if not os.path.isfile(tcl_file):
+        raise FileNotFoundError("{} not found.".format(tcl_file))
+
+    cmd = [
+        'vivado', '-quiet', '-nolog', '-notrace', '-mode', 'batch', '-source',
+        str(tcl_file), '-tclargs', part, bitstream
+    ]
+    try:
+        subprocess.run(cmd, check=True)
+    except FileNotFoundError as e:
+        print("Unable to execute Vivado: {}. Is Vivado in your PATH?".format(
+            str(e)))
+        return 1
+    except subprocess.CalledProcessError as e:
+        # We should have gotten an error message already from Vivado.
+        return 1
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--hw-target',
+                        '-t',
+                        required=False,
+                        default="",
```

Comment:
Personally, I think it's cleaner to pass `None` through when the argument isn't specified. This distinguishes between someone not passing the flag and someone passing `-t ''`: a silly error that can happen when called from a script which is probably better to explode on.